File: core/laboratory-tier/src/main/scala/ru/korus/tmis/laboratory/data/accept/ImageValue.scala
package ru.korus.tmis.laboratory.data.accept

import ru.korus.tmis.laboratory.data.lis.{accept => lab}
import ru.korus.tmis.laboratory.data.lis2.{accept => lab2}

final object ImageValue {
  implicit def fromLab1 (v: lab.ImageValue) = {
    ImageValue(Option(v.getDescription), Option(v.getImageData))
  }


  implicit def fromLab2 (v: lab2.ImageValue) = {
    ImageValue(Option(v.getDescription), Option(v.getImageData))
  }
}

/**
 * Класс для передачи изображений
 */
final case class ImageValue (
  /**
   * строка описания
   */
  description: Option[String],
  /**
   * картинка, закодированная в Base64
   */
  imageData: Option[String]
)

File: core/common/src/main/java/ru/korus/tmis/lis/data/ImageValue.java
package ru.korus.tmis.lis.data;

import javax.xml.bind.annotation.XmlElement;

/**
 * Класс для передачи изображений
 */
public class ImageValue {
    /**
     * строка описания
     */
    private String description;

    /**
     * картинка, закодированная в Base64
     */
    private String imageData;

    @XmlElement(name = "imageString")
    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @XmlElement(name = "imageData")
    public String getImageData() {
        return imageData;
    }

    public void setImageData(String imageData) {
        this.imageData = imageData;
    }
}
